tests/test_cases/test_sv_interface/test_sv_if.py

```diff
@@ -7,9 +7,50 @@
 
 @cocotb.test()
 async def test_sv_if(dut):
-    """Test that signals in an interface are discovered and iterable"""
+    """Test that signals in an interface are discovered"""
 
     dut.sv_if_i._discover_all()
     assert hasattr(dut.sv_if_i, "a")
     assert hasattr(dut.sv_if_i, "b")
     assert hasattr(dut.sv_if_i, "c")
+
+
+@cocotb.test()
+async def test_sv_intf_arr_type(dut):
+    """Test that interface arrays are the correct type"""
+
+    print(dut.sv_if_arr)
+
+    if cocotb.SIM_NAME.lower().startswith(("xmsim", "modelsim", "riviera")):
+        assert isinstance(dut.sv_if_arr, cocotb.handle.ArrayObject)
+    else:
+        # This is correct
+        assert isinstance(dut.sv_if_arr, cocotb.handle.HierarchyArrayObject)
+
+
+@cocotb.test(expect_fail=cocotb.SIM_NAME.lower().startswith("riviera"))
+async def test_sv_intf_arr_len(dut):
+    """Test that interface array length is correct"""
+    assert len(dut.sv_if_arr) == 3
+
+
+@cocotb.test(expect_fail=cocotb.SIM_NAME.lower().startswith("riviera"))
```

Comment:
`expect_error=IndexError if Riviera else ()`

Reply:
Ohh does expect_fail only cover asserts and not any general fail?

Reply:
That's correct. I'm glad you didn't know that, it's not a great system, and one we hope to replace, so users not touching them is a good thing.